ghc under gentoo prefix should use $EPREFIX/usr/bin/gcc
and not system's gcc. Otherwise system's gcc won't be able
to link stuff without feeding it with proper library paths.

Bug-url: https://bugs.gentoo.org/show_bug.cgi?id=384273
Reported-by: kazuhiro@rhodonit.com
diff --git a/aclocal.m4 b/aclocal.m4
index 65d4451..11bb003 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1617,15 +1617,7 @@ AC_DEFUN([XCODE_VERSION],[
 # --------------------------------
 # Finds where gcc is
 AC_DEFUN([FIND_GCC],[
-    if test "$TargetOS_CPP" = "darwin" &&
-        test "$XCodeVersion1" -ge 4
-    then
-        # From Xcode 4, use 'gcc-4.2' to force the use of the gcc legacy
-        # backend (instead of the LLVM backend)
-        FP_ARG_WITH_PATH_GNU_PROG([CC], [gcc-4.2])
-    else
-        FP_ARG_WITH_PATH_GNU_PROG([CC], [gcc])
-    fi
+    FP_ARG_WITH_PATH_GNU_PROG([CC], [gcc])
     export CC
     WhatGccIsCalled="$CC"
     AC_SUBST(WhatGccIsCalled)
